### PR TITLE
[Merged by Bors] - feat: bijective local homeomorphisms are homeomorphisms

### PR DESCRIPTION
**Pull request by grunweg**
Basis for generalising the result for local diffeomorphism to local structomorphisms.

### CODE REVIEW

File: Mathlib/Topology/IsLocalHomeomorph.lean [hunk 1]
```diff
@@ -205,6 +205,17 @@ theorem openEmbedding_of_injective (hf : IsLocalHomeomorph f) (hi : f.Injective)
     OpenEmbedding f :=
   openEmbedding_of_continuous_injective_open hf.continuous hi hf.isOpenMap
 
+/-- A surjective open embedding is a homeomorphism. -/
+noncomputable def _root_.OpenEmbedding.toHomeomeomorph_of_surjective (hf : OpenEmbedding f)
```
Comment (ADedecker) [inline Mathlib/Topology/IsLocalHomeomorph.lean:209]: I think you don't need to assume that the embedding is open, surjectivity implies it.

Comment (grunweg) [reply Mathlib/Topology/IsLocalHomeomorph.lean:209]: Indeed, good call. Replaced by just an embedding.

**Comment by grunweg** [comment]
Thanks for the quick look, @ADedecker!
I addressed the review comment. Sorry for the force-push; only the last comment is new.
awaiting-review

File: Mathlib/Topology/IsLocalHomeomorph.lean [hunk 2]
```diff
@@ -205,6 +205,17 @@ theorem openEmbedding_of_injective (hf : IsLocalHomeomorph f) (hi : f.Injective)
     OpenEmbedding f :=
   openEmbedding_of_continuous_injective_open hf.continuous hi hf.isOpenMap
 
+/-- A surjective embedding is a homeomorphism. -/
+noncomputable def _root_.Embedding.toHomeomeomorph_of_surjective (hf : Embedding f)
+    (hsurj : Function.Surjective f) : X ≃ₜ Y := by
+  apply Homeomorph.homeomorphOfContinuousOpen (Equiv.ofBijective f ⟨hf.inj, hsurj⟩)
```
Comment (ADedecker) [inline Mathlib/Topology/IsLocalHomeomorph.lean:211]:
```suggestion
    (hsurj : Function.Surjective f) : X ≃ₜ Y :=
  Homeomorph.homeomorphOfContinuousOpen (Equiv.ofBijective f ⟨hf.inj, hsurj⟩)
```

File: Mathlib/Topology/Maps.lean [hunk 3]
```diff
@@ -622,6 +622,11 @@ theorem openEmbedding_of_embedding_open {f : α → β} (h₁ : Embedding f) (h
   ⟨h₁, h₂.isOpen_range⟩
 #align open_embedding_of_embedding_open openEmbedding_of_embedding_open
 
+/-- A surjective embedding is an `OpenEmbedding`. -/
+theorem _root_.Embedding.toOpenEmbedding_of_surjective {f : α → β}
+    (hf : Embedding f) (hsurj: f.Surjective) : OpenEmbedding f :=
+  ⟨hf, (Set.range_iff_surjective.mpr hsurj) ▸ isOpen_univ⟩
```
Comment (ADedecker) [inline Mathlib/Topology/Maps.lean:628]:
```suggestion
  ⟨hf, hsurj.range_eq ▸ isOpen_univ⟩
```

**Review by ADedecker — APPROVED** [review]
Thanks!

bors d+

**Comment by grunweg** [comment]
Thank you for the fast review.
bors r+

**Pull request closed without merging**